OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -826,6 +826,34 @@ internal static void UpgradeWeaponRules(int engineVersion, ref List<MiniYamlNode
 					}
 				}
 
+				// Refactor Missile RangeLimit from ticks to WDist
+				if (engineVersion < 20160501)
+				{
+					var weapRange = node.Value.Nodes.FirstOrDefault(n => n.Key == "Range");
+					var projectile = node.Value.Nodes.FirstOrDefault(n => n.Key == "Projectile");
+
+					if (projectile != null && weapRange != null && projectile.Value.Value == "Missile")
+					{
+						var oldWDist = FieldLoader.GetValue<WDist>("Range", weapRange.Value.Value);
+						var rangeLimitNode = projectile.Value.Nodes.FirstOrDefault(x => x.Key == "RangeLimit");
+
+						// RangeLimit is now a WDist value, so for the conversion, we take weapon range and add 20% on top.
+						// Overly complicated calculations using Range, Speed and the old RangeLimit value would be rather pointless,
+						// because currently most mods have somewhat arbitrary, usually too high and in a few cases too low RangeLimits anyway.
+						var newValue = oldWDist.Length * 120 / 100;
+						var newCells = newValue / 1024;
+						var newCellPart = newValue - (newCells * 1024);
```

Comment:
Wait, I don't get this here... Above we say `newCells = newValue / 1024` so this here gets `newCellPart = newValue - ((newValue / 1024) * 1024) = newValue - newValue = 0`?